app/src/main/java/com/github/featuredetectandroid/ui/Menu.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.featuredetectandroid.ui
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.featuredetectandroid.ui.theme.Typography
+import com.github.featuredetectandroid.utils.algorithmsNamesAsList
+
+@Composable
+fun Menu(currentAlgorithm: String, changeSelectedAlgorithmIntPreferences: (String) -> Unit) {
+    val radioOptions = algorithmsNamesAsList()
+    val (selectedAlgorithm, onAlgorithmSelection) = remember { mutableStateOf(currentAlgorithm) }
```

Comment:
I don't really get what's going on here, explain, please:
1. How `currentAlgorithm`, being a `String`, is destructured into two values.
2. What do we need `onAlgorithmSelection` for?

Reply:
I used third way to declare `MutableState`: https://developer.android.com/jetpack/compose/state#state-in-composables

`selectedAlgorithm` is the selected value and `onAlgorithmSelection` is the setter of it. I pass `onAlgorithmSelection` as an argument into `onClickChangeSelectedAlgorithm` to be able to change `MutableState`'s value. `currentAlgorithm` is algorithm by deafault (= selected at the moment of `Menu(...)` invokation).

Reply:
But I can use something like that from their example: `onValueChange = { name = it }`

Reply:
Why you can't just do `val selectedAlgorithm by remember { mutableStateOf(currentAlgorithm) }` and set the value like `selectedAlgorithm = value`?

Reply:
> Why you can't just do `val selectedAlgorithm by remember { mutableStateOf(currentAlgorithm) }` and set the value like `selectedAlgorithm = value`?

I can do it. However, I don't know if this would be a more readable solution than passing the setter explicitly. Probably yes, though.

Reply:
For me it would be more readable if `onAlgorithmSelection` was named `setSelectedAlgorithm` as it's a setter.

Though, I believe you can get rid of `onClickChangeSelectedAlgorithm` all together if you remove `RadioButton`'s `onClick` like I said in other conversation.

Reply:
Fixed. About the `radioButton`, see below.